### PR TITLE
fixing putmask logic for scalar inputs

### PR DESCRIPTION
**Pull request by ipdemes**

### CODE REVIEW

**Comment by shriram-jagan** [comment]
LGTM! I can confirm that this fixes my use case. You could also optionally use `auto dim = std::max(1, args.in.dim());` like in `ScalarUnaryRedDispatch` to be consistent.

**Review by shriram-jagan — APPROVED** [review]
LGTM! I can confirm that this fixes my use case. You could also optionally use auto dim = std::max(1, args.in.dim()); like in ScalarUnaryRedDispatch to be consistent.

File: tests/integration/test_putmask.py [hunk 1]
```diff
@@ -67,6 +67,13 @@ def test_scalar():
     num.putmask(x_num, mask_num, values_num[:1])
     assert np.array_equal(x_num, x)
 
+    # the case when every input is a scalar
+    x = num.random.rand(3, 3)
+    s = x.sum()
+    num.putmask(s, True, 1.0)
+    print("IRINA DEBUG DDD", s)
```
Comment (shriram-jagan) [inline tests/integration/test_putmask.py:74]: The print statement somehow made it in....

**Review by manopapad — APPROVED** [review]

File: cunumeric/array.py [hunk 2]
```diff
@@ -4014,7 +4014,7 @@ def find_common_type(*args: Any) -> np.dtype[Any]:
                 scalar_types.append(array.dtype)
             else:
                 array_types.append(array.dtype)
-        return np.find_common_type(array_types, scalar_types)  # type: ignore
+        return np.find_common_type(array_types, scalar_types)
```
Comment (magnatelee) [inline cunumeric/array.py:4017]: I'd make sure this change doesn't make mypy unhappy.

File: src/cunumeric/index/putmask_template.inl [hunk 3]
```diff
@@ -105,7 +105,8 @@ static void putmask_template(TaskContext& context)
 {
   auto& inputs = context.inputs();
   PutmaskArgs args{context.outputs()[0], inputs[1], inputs[2]};
-  double_dispatch(args.input.dim(), args.input.code(), PutmaskImpl<KIND>{}, args);
+  int dim = args.input.dim() == 0 ? 1 : args.input.dim();
```
Comment (magnatelee) [inline src/cunumeric/index/putmask_template.inl:108]:
```suggestion
  int dim = std::max(1, args.input.dim());
```

File: cunumeric/array.py (hunk 2 above)
Comment (shriram-jagan) [reply cunumeric/array.py:4017]: @this is from #980. I'll let @ipdemes merge the changes in `putmask`

I think this change was needed to make sure mypy doesn't complain, so it does it make it happy. (@ipdemes would know more)

File: src/cunumeric/index/putmask_template.inl (hunk 3 above)
Comment (shriram-jagan) [reply src/cunumeric/index/putmask_template.inl:108]: I left the same comment in #980 :) @ipdemes , can you please fix it in your PR?

File: cunumeric/array.py (hunk 2 above)
Comment (ipdemes) [reply cunumeric/array.py:4017]: Yes, mypy was complaining about this `type: ignore` so I removed it

Comment (ipdemes) [reply cunumeric/array.py:4017]: Ha, mypy in CI complaining now. I will put it back

**Pull request merged**